### ISSUE TITLE
Error running pipeline on single sample

### ISSUE THREAD

**Issue by nhatas** [czbiohub/sc2-illumina-pipeline#68]
Can't seem to get the pipeline to run using Docker.

Command used:
`nextflow run czbiohub/sc2-illumina-pipeline -profile docker --reads 'data/*RNA_Stock9_TruSeq_QC_{R1,R2}.fq.gz' --kraken2_db 'data/kraken2/' --primers 'data/nCoV-2019.bed'`

Output:
```
N E X T F L O W  ~  version 20.07.1
Launching `czbiohub/sc2-illumina-pipeline` [modest_mclean] - revision: 532bb1ae0b [master]
NOTE: Your local project version looks outdated - a different revision is available in the remote repository [fbd944a8ba]
executor >  local (2)
executor >  local (3)
[-        ] process > prefilterHostReads                                -
[f4/1b2b21] process > quantifyERCCs (RNA_Stock9_TruSeq) [100%] 1 of 1 ✔
[da/88c5c0] process > filterReads (RNA_Stock9_TruSeq)   [  0%] 0 of 1
[-        ] process > trimReads (RNA_Stock9_TruSeq)     -
[-        ] process > alignReads                                        -
[-        ] process > trimPrimers                                       -
[-        ] process > makeConsensus                                     -
[-        ] process > quast                                             -
[-        ] process > callVariants                                      -
[-        ] process > filterVariants                                    -
[-        ] process > computeStats                                      -
[-        ] process > combinedVariants                                  -
[-        ] process > mergeAllAssemblies                                -
[-        ] process > mergeAssemblyStats                                -
[-        ] process > filterAssemblies                                  -
[-        ] process > snpsDf                                            -
[-        ] process > qcTreeMap                                         -
[-        ] process > multiqc                                           -
Error executing process > 'filterReads (RNA_Stock9_TruSeq)'

Caused by:
  Cannot invoke method complete() on null object
```

**Comment by jackkamm** [comment]
Hi,

Can you try running like this and see if it works:
```
nextflow run czbiohub/sc2-illumina-pipeline -profile artic,docker --reads 'data/*RNA_Stock9_TruSeq_QC_{R1,R2}.fq.gz' --kraken2_db 'data/kraken2/'
```
Also, could you let me know how it looks when you `ls data/kraken2/`?

Finally, please attach a copy of `.nextflow.log` if it still doesn't work.

**Comment by nhatas** [comment]
The contents of the kraken2 directory are:
`hash.k2d  kraken2_h+v_20200319.tar.gz  opts.k2d  taxo.k2d`

I had to modify the command a little bit because I had to change directories. The command I ran is nextflow run `czbiohub/sc2-illumina-pipeline -profile artic,docker --reads '/mnt/pathogen1/stahan/_old/M622/cz_test/*RNA_Boon_Stock9_TruSeq_QC_{R1,R2}.fq.gz' --kraken2_db '/mnt/pathogen1/stahan/_old/M622/kraken2/'`

I have attached the .nextflow.log file
[nextflow_log.zip](https://github.com/czbiohub/sc2-illumina-pipeline/files/5457270/nextflow_log.zip)

**Comment by jackkamm** [comment]
From the log it looks like it made it further this time to the `combinedVariants` step but crashed then.

It appears to be a bug when running the pipeline on only a single sample. I'm able to reproduce the error. We always run on several samples at a time which is why I missed this before.

I'll fix the error soon. In the meantime a workaround would be to add a couple more samples to the run. If you don't have good samples to use on hand, you could grab some of these samples from our test suite:
```
https://github.com/czbiohub/test-datasets/raw/msspe/testdata/empty_R{1,2}.fastq.gz
https://github.com/czbiohub/test-datasets/raw/msspe/testdata/sample1_R{1,2}_001.fastq.gz
https://github.com/czbiohub/test-datasets/raw/olgabot/msspe--add-human-and-kraken/testdata/RR057e_00734_subsampled_R{1,2}.fastq.gz
```

**Comment by nhatas** [comment]
I downloaded one of the datasets you linked to my work directory. This is the contents of my work directory (`/mnt/pathogen1/stahan/_old/M622/cz_test/'):

```
drwxrwsr-x  6 stahan stahan      4096 Nov  9 15:07 .
drwxrwsr-x 10 stahan stahan      4096 Oct 22 19:17 ..
-rw-rw-r--  1 stahan stahan 552562956 Oct 12 17:07 M622_I7929_RNA_Boon_Stock9_TruSeq_QC_R1.fq.gz
-rw-rw-r--  1 stahan stahan 571546880 Oct 12 17:07 M622_I7929_RNA_Boon_Stock9_TruSeq_QC_R2.fq.gz
-rwx--x--x  1 stahan stahan     15478 Oct 29 02:50 nextflow
drwxrwsr-x  3 stahan stahan      4096 Nov  9 15:08 .nextflow
-rw-rw-r--  1 stahan stahan     25170 Nov  9 15:08 .nextflow.log
-rw-rw-r--  1 stahan stahan     23128 Nov  9 15:04 .nextflow.log.1
-rw-rw-r--  1 stahan stahan     32607 Oct 29 04:33 .nextflow.log.2
drwxrwsr-x  3 stahan stahan      4096 Nov  9 15:02 null
drwxrwsr-x 11 stahan stahan      4096 Nov  9 15:08 out
-rw-rw-r--  1 stahan stahan     83853 Nov  9 15:00 sample1_R1_001.fastq.gz
-rw-rw-r--  1 stahan stahan     83829 Nov  9 15:00 sample1_R2_001.fastq.gz
drwxrwsr-x 29 stahan stahan      4096 Nov  9 15:08 work
```

I am still getting an error when running the pipeline. The command I used was: 
```
nextflow run czbiohub/sc2-illumina-pipeline -profile artic,docker --reads '/mnt/pathogen1/stahan/_old/M622/cz_test/*_{R1,R2}_*.gz' --kraken2_db '/mnt/pathogen1/stahan/_old/M622/kraken2/' --outdir '/mnt/pathogen1/stahan/_old/M622/cz_test/out/'
```

[20201109_nextflow.log](https://github.com/czbiohub/sc2-illumina-pipeline/files/5513085/20201109_nextflow.log)

**Comment by jackkamm** [comment]
Sorry for the delay. This should now be fixed by #69. Please reopen if it still doesn't work.
